lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1932,6 +1932,11 @@ void AttributeChecker::visitSpecializeAttr(SpecializeAttr *attr) {
 }
 
 void AttributeChecker::visitFixedLayoutAttr(FixedLayoutAttr *attr) {
+  if (isa<StructDecl>(D)) {
+    TC.diagnose(attr->getLocation(), diag::fixed_layout_struct)
+      .fixItReplace(attr->getRange(), "@frozen");
```

Comment:
I think this needs to be `getRangeWithAt` (or that you should omit the `@` from your replacement).

Reply:
I think I'm missing something, this looks right as-is to me:
```
'@frozen' attribute is now used for fixed-layout structs
@_fixed_layout
^~~~~~~~~~~~~~
@frozen
```

Reply:
Are you sure you weren't seeing

```
'@frozen' attribute is now used for fixed-layout structs
@_fixed_layout
^~~~~~~~~~~~~~
 @frozen
```

? Which is a different thing.

Reply:
A `@_fixed_layout public struct FixedDeprecationWarning {} // expected-warning {{'@frozen' attribute is now used for fixed-layout structs}} {{1-15=@frozen}}` test passes.